sdk/storage/storage-blob/src/BlobServiceClient.ts

```diff
@@ -790,6 +790,7 @@ export class BlobServiceClient extends StorageClient {
   public listContainers(
     options: ServiceListContainersOptions = {}
   ): PagedAsyncIterableIterator<ContainerItem, ServiceListContainersSegmentResponse> {
+    if (options.prefix === "") options.prefix = undefined;
```

Comment:
Hey, @sanallur@microsoft.com 

Would you share more detail about the context "request is bad", is it caused by the protocol layer?(i.e. generated code)

If so, what's the behavior of protocol layer when "" is provided for options.prefix in protocol layer.

My idea here is: As my understanding, protocol layer should also be usable SDK, so if the issue is sourced from protocol layer, we might consider to fix it in that layer.

Reply:
[From #5817]
When the prefix is a null string, the request goes through with this kind of url - `https://myaccount.blob.core.windows.net/?prefix=&comp=list` 
and the service throws an error as follows.

```bash
 Error: ﻿<?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is 
formed correctly including the signature.
RequestId:0fd08c11-d01e-003b-128c-9324a2000000
Time:2019-11-05T03:53:30.8126068Z</Message><AuthenticationErrorDetail>The MAC signature found in the HTTP request 'INY/ACxexL3YBrpiEx9TYry6WDcfvBisH/Ohl2rPc=' is not the same as any computed signature. Server used following string to sign: 'GET
x-ms-client-request-id:162c4131-5921-4d22-b7e1-1bb1b705ccf7
x-ms-date:Tue, 05 Nov 2019 03:52:57 GMT
x-ms-version:2019-02-02
/myaccount/
comp:list
prefix:'.</AuthenticationErrorDetail></Error>
```

Reason being the value of the prefix attribute is an empty string.
Now, I'm not sure who should handle this case
- SDK (like this PR) - override when the `prefix` attribute is an empty string
- core-http - handle the case when a value of any query parameter is an empty string
- Service should realize that this is a valid url and provide response accordingly

Reply:
cc - @daviwil

Reply:
We should fix this in convenience layer, this issue likelys relates with sharedkey authentication.

Reply:
Talked with @daviwil offline if core-http should sanitize. He seems to like the fix in this PR.